apps/prairielearn/assets/scripts/instructorAiGenerateDraftEditorClient.ts

```diff
@@ -0,0 +1,85 @@
+import ace from 'ace-builds';
+import { observe } from 'selector-observer';
+
+import { onDocumentReady } from '@prairielearn/browser-utils';
+
+import { configureAceBasePaths } from './lib/ace.js';
+import { saveButtonEnabling } from './lib/saveButtonEnabling.js';
+
+class DraftFileEditor {
```

Comment:
A lot of this is borrowed from the `InstructorFileEditor` class on the other file editor page. I didn't yet go through the trouble of trying to make that flexible enough to be reused here, though we may want to do that in the future.